test/BlankNodeManagerTest.cpp

```diff
@@ -40,21 +40,21 @@ TEST(BlankNodeManager, LocalBlankNodeManagerGetID) {
   BlankNodeManager::LocalBlankNodeManager l(&bnm);
 
   // initially the LocalBlankNodeManager doesn't have any blocks
-  EXPECT_EQ(l.blocks_.size(), 0);
+  EXPECT_EQ(l.blocks_->size(), 0);
 
   // A new Block is allocated, if
   // no blocks are allocated yet
   uint64_t id = l.getId();
-  EXPECT_EQ(l.blocks_.size(), 1);
+  EXPECT_EQ(l.blocks_->size(), 1);
   EXPECT_TRUE(l.containsBlankNodeIndex(id));
   EXPECT_FALSE(l.containsBlankNodeIndex(id + 1));
   EXPECT_FALSE(l.containsBlankNodeIndex(id - 1));
 
   // or the ids of the last block are all used
-  l.blocks_.back().nextIdx_ = id + BlankNodeManager::blockSize_;
+  l.blocks_->back().nextIdx_ = id + BlankNodeManager::blockSize_;
   id = l.getId();
   EXPECT_TRUE(l.containsBlankNodeIndex(id));
-  EXPECT_EQ(l.blocks_.size(), 2);
+  EXPECT_EQ(l.blocks_->size(), 2);
```

Comment:
Add a test, that the accidental recursive merging works properly (if a blank node manager recursively contains itself).